IS1001Test.py

```diff
@@ -12,12 +12,623 @@
+            ))
+        if "location" not in response.headers:
+            raise NMOSTestException(test.FAIL(
+                "'Location' not found in Response headers when requesting '{}'. Headers = {}"
+                .format(query_key, response.headers.keys())
+            ))
+        if query_key not in response.headers["location"]:
+            raise NMOSTestException(test.FAIL(
+                "'{}' not found in Location Header. Location Header = {}"
+                .format(query_key, response.headers["location"])
+            ))
+        if "state" not in response.headers["location"]:
+            raise NMOSTestException(test.FAIL(
+                "'state' not found in Location Header. Location Header = {}"
+                .format(response.headers["location"])
+            ))
+
+        redirect_uri, query_string = response.headers["location"].split('?')
+        parsed_query = parse_qs(query_string)
+        actual_query_value = parsed_query[query_key][0]
+        if "state" in parsed_query.keys():
+            state = parse_qs(query_string)["state"][0]
+        else:
+            state = None
+
+        if not redirect_uri == input_params["redirect_uri"]:
+            raise NMOSTestException(test.FAIL(
+                "Expected {} but got {}".format(input_params["redirect_uri"], redirect_uri)
+            ))
+
+        if not state == input_params["state"]:
+            raise NMOSTestException(test.FAIL(
+                "Expected {} but got {}".format(input_params["state"], state)
+            ))
+
+        if query_key != "code":
+            if expected_query_value is None:
+                if not actual_query_value == input_params[query_key]:
+                    raise NMOSTestException(test.FAIL(
+                        "Expected {} but got {}".format(input_params[query_key], actual_query_value)
+                    ))
+            else:
+                if not actual_query_value == expected_query_value:
+                    raise NMOSTestException(test.FAIL(
+                        "Expected {} but got {}".format(expected_query_value, actual_query_value)
+                    ))
+        else:
+            return actual_query_value
+
+    def do_dns_sd_advertisement_check(self, test, api, service_type):
+        """Auth API advertises correctly via mDNS"""
+
+        if DNS_SD_MODE != "multicast":
+            return test.DISABLED("This test cannot be performed when DNS_SD_MODE is not 'multicast'")
+        print(api)
+
+        ServiceBrowser(self.zc, service_type, self.zc_listener)
+        sleep(DNS_SD_BROWSE_TIMEOUT)
+        serv_list = self.zc_listener.get_service_list()
+        for service in serv_list:
+            address = socket.inet_ntoa(service.address)
+            port = service.port
+            if address == api["ip"] and port == api["port"]:
+                properties = self.convert_bytes(service.properties)
+                if "pri" not in properties:

```

Comment:
TXT records don't seem to be specified in IS-10 at present. They almost certainly should be, although perhaps not `api_proto` as OAuth without TLS isn't ideal.